src/test/java/apicurito/tests/configuration/templates/ApicuritoTemplate.java

```diff
@@ -76,37 +76,44 @@ public static void deployUsingTemplate() {
     private static void deployUsingGoOperator() {
         log.info("Deploying using GO operator");
 
-        deployOnOCP("CRD", TestConfiguration.apicuritoOperatorCrdUrl());
-        deployOnOCP("Service", TestConfiguration.apicuritoOperatorServiceUrl());
-        deployOnOCP("Role", TestConfiguration.apicuritoOperatorRoleUrl());
-        deployOnOCP("Role binding", TestConfiguration.apicuritoOperatorRoleBindingUrl());
-        deployOnOCP("Operator", TestConfiguration.apicuritoOperatorUrl());
+        createInOCP("CRD", TestConfiguration.apicuritoOperatorCrdUrl());
+        createInOCP("Service", TestConfiguration.apicuritoOperatorServiceUrl());
+        createInOCP("Role", TestConfiguration.apicuritoOperatorRoleUrl());
+        createInOCP("Role binding", TestConfiguration.apicuritoOperatorRoleBindingUrl());
+        createInOCP("Operator", TestConfiguration.apicuritoOperatorUrl());
 
-        deployCr(TestConfiguration.apicuritoOperatorCrUrl());
+        applyInOCP("Custom Resource", TestConfiguration.openShiftNamespace(), TestConfiguration.apicuritoOperatorCrUrl());
+
+        if (TestConfiguration.apicuritoOperatorUiImage() != null) {
+            setTestEnvToOperator("APICURITO_IMAGE", TestConfiguration.apicuritoOperatorUiImage());
+        }
     }
 
-    private static void deployOnOCP(String itemName, String item) {
-        log.info("Deploying {} from: {}", itemName, item);
+    private static void setTestEnvToOperator(String nameOfEnv, String valueOfEnv) {
+        log.info("Setting test ENV: " + nameOfEnv + "=" + valueOfEnv);
         final String output = OpenShiftUtils.binary().execute(
-                "create",
-                "-n", TestConfiguration.openShiftNamespace(),
-                "-f", item
+            "set",
+            "env",
+            "deployment",
+            "apicurito-operator",
+            nameOfEnv + "=" + valueOfEnv
         );
     }
 
-    public static void applyOnOCP(String itemName, String namespace, String item) {
-        log.info("Applying {} from: {}", itemName, item);
+    private static void createInOCP(String itemName, String item) {
+        log.info("Creating " + itemName + " from: " + item);
+
         final String output = OpenShiftUtils.binary().execute(
-                "apply", "-n", namespace, "-f", item
+            "create",
+            "-n", TestConfiguration.openShiftNamespace(),
+            "-f", item
         );
     }
 
-    public static void deployCr(String cr) {
-        log.info("Deploying CR from " + cr);
+    public static void applyInOCP(String itemName, String namespace, String item) {
```

Comment:
There I was just thinking about "applyInOCP" with 2 parameters, with specified default namespace. Just a "nitpicky" idea, not really necessary, but it could be used in several places - basically similar to "createInOCP" method above

Reply:
applyInOCP must contain an option to specify namespace because of operatorhub test where something must be applied in "openshift-marketplace" namespace and something in regular namespace i.e. "operatorhub" or "apicurito"